### PR TITLE
Implemented Validation for .tar.gz pypi source package

### PR DESCRIPTION
**Pull request by anki2189**
# Pull Request Description

This pull request closes #

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [x] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

### CODE REVIEW

**Comment by anki2189** [comment]
@strongbox/core-developers : Please review this PR.
`.tar.gz` package can have `-` in its name. Current code is splitting artifactName on `-` to extract distribution/package name which is not correct.

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/util/PypiArtifactCoordinatesUtils.java [hunk 1]
```diff
@@ -14,6 +18,14 @@
 public class PypiArtifactCoordinatesUtils
 {
 
+    private static Pattern PACKAGE_VERSION_PATTERN = Pattern.compile(
+                                                                     PypiPackageInfo.VERSION_FORMAT,
+                                                                     Pattern.CASE_INSENSITIVE);
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/util/PypiArtifactCoordinatesUtils.java]: Cosmetic: Could you please move the first parameter to the same line? (Here and below).
```java
    private static Pattern PACKAGE_VERSION_PATTERN = Pattern.compile(PypiPackageInfo.VERSION_FORMAT,
                                                                     Pattern.CASE_INSENSITIVE);
```

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/util/PypiArtifactCoordinatesUtils.java [hunk 2]
```diff
@@ -14,6 +18,14 @@
...
+
+    private static Pattern PACKAGE_DISTRIBUTION_NAME_PATTERN = Pattern.compile(
+                                                                               PypiPackageInfo.DISTRIBUTION_NAME_FORMAT,
+                                                                               Pattern.CASE_INSENSITIVE);
+
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/util/PypiArtifactCoordinatesUtils.java]:
```java
    private static Pattern PACKAGE_DISTRIBUTION_NAME_PATTERN = Pattern.compile(PypiPackageInfo.DISTRIBUTION_NAME_FORMAT,
                                                                               Pattern.CASE_INSENSITIVE);
```

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/util/PypiArtifactCoordinatesUtils.java (hunk 1 above)
Comment (anki2189) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/util/PypiArtifactCoordinatesUtils.java]: Fixed

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/util/PypiArtifactCoordinatesUtils.java (hunk 2 above)
Comment (anki2189) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/util/PypiArtifactCoordinatesUtils.java]: Fixed.

**Pull request merged**